R/target_market_share.R

```diff
@@ -437,11 +437,23 @@ reweight_technology_share <- function(data, ...) {
     group_by(...) %>%
     mutate(
       .x = .data$weighted_technology_share,
+      .sum_x = sum(.data$.x),
       .y = .data$weighted_technology_share_target,
-      weighted_technology_share = .data$.x / sum(.data$.x),
-      weighted_technology_share_target = .data$.y / sum(.data$.y),
+      .sum_y = sum(.data$.y),
+      weighted_technology_share = ifelse(
+        .data$.sum_x == 0,
+        0,
+        .data$.x / .data$.sum_x
+        ),
```

Comment:
Nit: Is it worth taking so much vertical screen-real-state?

Reply:
Nope! You are right, this was silly.